app/models/name/spelling.rb

```diff
@@ -75,9 +75,13 @@ def self.names_from_synonymous_genera(str, parent = nil)
       # synonym = <Lepiota>
       parent.synonyms.each do |synonym|
         # "Lepiota bog% var. nam%"
-        conditions = ["text_name like ? AND correct_spelling_id IS NULL",
-                      "#{synonym.text_name} #{child_pat}"]
-        result += Name.where(conditions).select do |name|
+        # conditions = ["text_name like ? AND correct_spelling_id IS NULL",
+        #               "#{synonym.text_name} #{child_pat}"]
+        # result += Name.where(conditions).select do |name|
+        result += Name.where(
+          Name[:text_name].matches("#{synonym.text_name} #{child_pat}").
+          and(Name[:correct_spelling_id].eq(nil))
+        ).select do |name|
```

Comment:
Again, more AR and less Arel looks nicer to me:

`Name.where(Name[:text_name].matches(...)).where(correct_spelling: nil).select...`